content/blog/enforcing-and-automatically-tagging-aws-resources/index.md

```diff
@@ -0,0 +1,745 @@
+---
+date: "2020-04-03"
+title: "Enforcing and Automatically Tagging AWS Resources"
+authors: ["joe-duffy"]
+tags: ["AWS", "policy-as-code"]
+meta_desc: "Tagging resources is an AWS best practice. In this post, learn how to enforce, and easy wasys to automate, tagging your infrastructure reliably."
+meta_image: "pac-tags-meta.png"
+---
+
+AWS publishes best practices for how to tag your resources for cost tracking, automation, and organization. But how do you enforce that you're doing it correctly across all of your projects? And is it really necessary to manually track down all those places where you missed a tag and manually patch things up? In this article, we'll see how to use Policy as Code to enforce your team's tagging strategies in addition to some powerful Infrastructure as Code techniques to automate applying your tags in a consistent way across all of your projects and resources.
+
+## Why Tag Your Resources?
+
+A tag is simply a key/value label that you can apply to your AWS infrastructure resources. This enables you to manage, search for, and filter resources. There aren't any predefined tags &mdash; you can use whatever makes sense for your scenario and business requirements.
+
+[Amazon recommends a number of tagging strategies](https://aws.amazon.com/answers/account-management/aws-tagging-strategies/), including technical tags like name and environment, automation tags like dates and security requirements, business tags like owner and cost center, and security tags for compliance. Each of these enables you to apply policies.
+
+Specifying a tag in your infrastructure as code is easy. Not all resources are taggable (although the most important ones are); to tag a resource, simply specify a map of key/values using the tags property. For example, this code declares an S3 Bucket that carries three tags that enable cost allocation reporting: `"user:Project"`, `"user:Stack"`, and `"user:Cost Center"`:
+
+{{< langchoose csharp >}}
```

Comment:
I think the plan is to kill off the old `langchoose` as soon as some remaining usage in the generated API docs have been expunged, so probably best for new content to use the new chooser: https://github.com/pulumi/docs/tree/master/components#pulumi-chooser

Reply:
Yes, thanks -- as soon as the docs have all been updated, I'll be removing these.